iaia/magicmodule.py

```diff
@@ -56,27 +59,45 @@ def __repr__(self):
     def __call__(self, *args, **kw):
         key = self.call_key(*args, **kw)
         if key not in self.funcs:
-            self.make_function(*args, **kw)
+            self.make_functions(*args, **kw)
         exc = None
-        try:
-            return self.funcs[key](*args, **kw)
-        except Exception as e:
-            exc = e
-        print(f"Attempting to fix exception {exc}...")
-        self.fix_function(exc, *args, **kw)
-        return self.funcs[key](*args, **kw)
+        for func_i, func in self.funcs[key].items():
+            try:
+                return func(*args, **kw)
+            except Exception as e:
+                exc = e
+            print(f"Attempting to fix exception {exc} in func {func_i}...")
+            func = self.fix_function(func_i, exc, *args, **kw)
+            return func(*args, **kw)
 
     def call_key(self, *args, **kw):
         return tuple([len(args), *sorted(kw.keys())])
 
+    def format_type_for_prompt(self, arg):
+        """
+        Format arguments for the prompt, adding info about function args
+        """
+        name = type(arg).__name__
+        if name == "function":
+            arg_count = arg.__code__.co_argcount
+            return "Callable[" + ", ".join(["Any"] * arg_count) + "]"
+        return name
+
+    def format_signature_for_compile(self, signature: str):
+        """
+        Format function signature so it's compilable by Python
+        """
+        return re.sub(": Callable\[.*\]", "", signature)
```

Comment:
What's the error that this is being used to remove? Lack of `Callable` in globals? (We could import some stuff into `self.ns`)

Reply:
right, `Callable` doesn't exist. Original error was "`function` does not exist", since that was the original arg type of functions before I forked the repo. Adding `typing` as an import